icaruscode/PMT/Trigger/trigger_emulation_icarus.fcl

```diff
@@ -363,19 +363,79 @@ icarus_standard_triggersim: {
       EmitEmpty:             true
       ExtraInfo:             true
     }
-    
+
   } # producers
-  
-  path: [ 
+
+  path: [
     #  pmtfixedthr  # this should be already in the input (or in the workflow!)
       pmtlvdsgates
     , pmttriggerwindows
     , triggersimgates
     , emuTrigger
   ]
-  
+
 } # icarus_standard_triggersim
 
+icarus_shifting_triggersim: {
+
+  producers: {
+
+    pmtfixedthrinit: @local::icarus_pmtdiscriminatethr_fixed_MC
+
+    pmtlvdsgatesinit: {
+      @table::icarus_standard_triggersim.producers.pmtlvdsgates
+      TriggerGatesTag: pmtfixedthrinit
+    }
+
+    pmttriggerwindowsinit: {
+      @table::icarus_standard_triggersim.producers.pmttriggerwindows
+      TriggerGatesTag: pmtlvdsgatesinit
+    }
+
+    pmttriggerwindowsinit: {
+      @table::icarus_standard_triggersim.producers.pmttriggerwindows
+      TriggerGatesTag: pmtlvdsgatesinit
+    }
+
+    triggersimgatesinit: {
+      @table::icarus_standard_triggersim.producers.triggersimgates
+    }
+
+    emuTriggerUnshifted: {
+      @table::icarus_standard_triggersim.producers.emuTrigger
+      BeamGates: triggersimgatesinit
+      TriggerGatesTag: pmttriggerwindowsinit
+    }
+
+    shifted: {
+       module_type: AdjustSimForTrigger
+       InputTriggerLabel: "emuTriggerUnshifted"
+       InitAuxDetSimChannelLabel: "largeant"
+       InitBeamGateInfoLabel: "triggersimgatesinit"
+       InitSimEnergyDepositLabel: "ionization"
+       InitSimPhotonsLabel: "largeant"
+       InitWaveformLabel: "opdaq"
+       ShiftAuxDetIDEs: true
+       ShiftBeamGateInfo: true
```

Comment:
Can you remind me the rationale behind this?
```
       ShiftBeamGateInfo: true
```

I remember having discussed that the beam gate object (`sim::BeamGateInfo`) is on simulation time scale and therefore should not be shifted, but the issue was probably more involved than just this.

Reply:
If I understood your previous advice correctly, I am shifting the `sim::BeamGateInfo` produced by `triggersimgatesinit`, which is then provided `emuTrigger` (the shifted one). I changed the input labels and give `triggersimgates` a `DummyProducer` [here](https://github.com/SBNSoftware/icaruscode/pull/673/files#diff-d3f990401fb3e7d310310cc20642a4b0cb120f1301e3bb0b17a4eb1f1104835fR71-R72).

Reply:
Thank you for providing me some output to better understand what happens.

In principle, `sim::BeamGateInfo` is expected to be on simulation time scale (something that says `0` when the generator is ready to put signal in), which for neutrino samples is usually the time of arrival of the spill. So our generators put in the `generator` data product a gate that goes from -0.2 µs to say 2.0 µs (to make up for a 2.2 µs BNB gate). At the beginning of digitisation stage, this simulation time scale still matches the electronics time scale, which is what everything detected and reconstructed is based on (this is the scale that is trigger-based in data and in simulation was "beam-gate" based).
The module `TriggerSimulationOnGates` runs trigger emulation on a input "beam" gate, and it needs that gate in the optical time scale, which is the time scale the PMT waveforms are on. So that module converts the input gate into optical time scale assuming that
1. the input gate is on simulation time scale, because that's the standard;
2. the optical time scale the PMT waveforms are based on corresponds to the electronics time scale (de facto convention, and enforced by data decoding);
3. the electronics time scale starts with the simulation time scale (barred the offset `1500`), because there is no place in LArSoft where their relation is stored and this one is the "standard" practice.

This gate conversion results therefore trivial (like, shifting ahead of `1500` µs).

Now the configuration in this PR has changed the practice at point (3) so that the electronics time scale, and the optical one, match the trigger time scale, not the simulation time scale any more. The old gate conversion of `TriggerSimulationOnGates` won't suffice any more.

The ~solution~ _workaround_ in this PR is to break the standard at point (1) and provide `TriggerSimulationOnGates` a gate that, after the trivial transformation, encloses the appropriate time interval. This gate is in a "physics interaction" time scale (making it up here and now) that has the 0 at the time of the interaction (allegedly equivalent to the one of the trigger).

The module `TriggerSimulationOnGates` in `v09_89_01_01` (and probably earlier) has an additional degree of freedom: it allows to specify which time scale the input gate is in. This should allow the module to perform the appropriate transformation to optical time scale. So the _solution_ to the issue would be to use the new module with the appropriate time reference in the configuration (which should be something like `BeamGate` or `Simulation`). In practice,
1. this PR has no access to that new code
2. it is not guaranteed that the new code has already the needed conversion among the available options
3. furthermore, additional test would be needed to verify that the conversion feature works as expected, since it's quite new and it may need more testing.

The only thing we need to pay attention to is that with the new `TriggerSimulationOnGates`, the default beam gate reference reproduces the behaviour of the old one.